:recycle: Refactors test_urlrequest.py

AndreMiras opened the pull request:
- uses @pytest.mark.skipif() rather than return
- couple of DRY fixes using helper functions

AndreMiras commented on kivy/tests/test_urlrequest.py:
```diff
@@ -5,17 +5,50 @@
+class UrlRequestQueue:
 
     def __init__(self, queue):
-        super(UrlRequestQueue, self).__init__()
```
Removed because we don't inherit from anything

AndreMiras commented on kivy/tests/test_urlrequest.py:
```diff
@@ -5,17 +5,50 @@
 
-class UrlRequestQueue(object):
 
-    queue = []
```
The `queue` is already defined in the init, less is more. Having it defined here can lead to the misleading assumption we want it to be static attribute

AndreMiras commented on kivy/tests/test_urlrequest.py:
```diff
@@ -5,17 +5,50 @@
+    timed_out = False
+    while not request.is_finished and not timed_out:
+        Clock.tick()
+        sleep(.1)
```
reduced the sleep time from .5 to .1 because nobody likes to wait :)

AndreMiras commented:
A small test refactoring before addressing #6946, also refs https://discord.com/channels/423249981340778496/490493814281338890/722883388889104446

matham commented on kivy/tests/test_urlrequest.py:
```diff
@@ -5,17 +5,50 @@
 
+if not os.environ.get('NONETWORK'):
+    from kivy.network.urlrequest import UrlRequest
+    from kivy.clock import Clock
```
Can you use the `kivy_clock` fixture and pass it in, instead of importing it? 

Also, I don't like when we import kivy things globally in tests because then the kivy stuff is initialized before the tests start because kivy imports are so aggressive. It'd be better if you just do `from kivy.network.urlrequest import UrlRequest` in each test.

AndreMiras replied:
I've updated the code as per your suggestion.
Out of curiosity, regarding kivy imports being aggressive. Dont't we have a deeper problem we want to fix one day? Or is it a feature that can't be fixed or that we don't want to fix?

matham replied:
Yeah, we do have a very deep problem there that I hope will be fixed one day (the sooner the better). So this is just in preparation for that bright sunny morning :D

AndreMiras commented:
Addressed all the comments, including:
> What's up with the :recycle: in the title?

It's commit emoji, https://gitmoji.carloscuesta.me/ I assume you didn't like it, so I removed it :smile:

matham commented:
Oh, it didn't render as a ♻️ in the title so I was confused by what it was supposed to be.

AndreMiras commented:
Yes, BTW, I just see the merge message picked the GitHub title rather than the actual commit title. So now we have a merge commit with the icon :man_facepalming:
